### PR TITLE
Stats: Add period dropdown to chart header

### PR DESCRIPTION
**Pull request by dognose24**
Related to https://github.com/Automattic/red-team/issues/251

## Proposed Changes

* Prevent focusing on the first option when launching the period dropdown.
* Add `IntervalDropdown` to the date filtering dedicated `ChartHeader`.
* Simply condition to display legend under the feature flag `stats/new-date-filtering`.

## Why are these changes being made?

* These changes introduce the `IntervalDropdown` to `ChartHeader`, which is dedicated to the new feature of showing along with the chart legend.

## Testing Instructions

* Spin this change up with the Calypso Live branch.
* Ensure the chart period selection works as previously.
* Apply the feature flag to the URL: `?flags=stats/new-date-filtering`.
* Ensure the period dropdown display next to the chart legend is aligned with the design.

|Before|After|
|-|-|
|截圖 2024-11-16 凌晨12 42 45|截圖 2024-11-18 凌晨12 02 44|

* Ensure the change does not break gating period options.

截圖 2024-11-17 晚上11 39 07

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/hooks/use-intervals.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		statType: STATS_FEATURE_INTERVAL_DROPDOWN_MONTH,
+	},
+	[ STATS_PERIOD.YEAR ]: {
+		id: STATS_PERIOD.YEAR,
+		label: translate( 'Years' ),
+		statType: STATS_FEATURE_INTERVAL_DROPDOWN_YEAR,
+	},
+};
+
+function useIntervals( siteId: number | null ): IntervalsType {
```
Comment (kangzj) [inline client/my-sites/stats/hooks/use-intervals.tsx:46]: Nice refactoring!

File: client/components/stats-interval-dropdown/index.jsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ const StatsIntervalDropdownListing = ( { selected, onSelection, intervals, onGat
 				},
 				{
 					name: 'jetpack_stats_upsell_clicked',
-					params: { statType: intervals[ interval ].statType, source: event_from },
+					params: { stat_type: intervals[ interval ].statType, source: event_from },
```
Comment (kangzj) [inline client/components/stats-interval-dropdown/index.jsx:26]: Is this an intentional change? It might break consistency in stats data, wouldn't it?

**Review by kangzj — APPROVED** [review]
It looks good but it doesn't seem to respect the date range? Please see video below:


https://github.com/user-attachments/assets/ee235c9a-a173-4eca-955b-74f090c963f3


It doesn't affect production, so I'm happy for it to merge and iterate. Nice work 👍

**Comment by annacmc** [comment]
I love it, looks great! I am seeing the same troubles as @kangzj - and it's also highlighted some other weirdness with the datepicker for me. But I am certain this is not caused by your PR, where I would expect clicking in the date field textbox would then allow me to select that date - but it isn't the case. I made a video here but it's a little hard to see what I'm trying to do:


https://github.com/user-attachments/assets/7a0dd19e-bd19-4d85-8f32-c5f3c60c92b4

But I think what you want to achieve works well, and the rest of the issues to do with the datepicker weirdness can be addressed separately 😄 🚀

**Review by annacmc — APPROVED** [review]

File: client/my-sites/stats/stats-chart-tabs/chart-header.jsx [hunk 3]
```diff
@@ -1,34 +1,65 @@
...
+				availableCharts={ availableLegend }
+				clickHandler={ onLegendClick }
+				tabs={ charts }
+			/>
+			<IntervalDropdown
+				slug={ slug }
+				period={ period }
+				queryParams={ queryParams }
+				intervals={ intervals }
+				onGatedHandler={ onGatedHandler }
```
Comment (kangzj) [inline client/my-sites/stats/stats-chart-tabs/chart-header.jsx:54]: I guess we need a `onSelection` handler connected back to the original logic, but I'm happy for it to be addressed in follow up PRs.

**Comment by kangzj** [comment]
I merged it as I'll have a prototype for slots refactoring around relevant code, but I don't think which is dead important really.

cc @annacmc

**Pull request merged**